File: Data Structures & Algorithms/valid-palindrome-ii/submission-2.py
class Solution:
    def validPalindrome(self, s: str) -> bool:

        def check_palindrome(low,high):

            while low<=high:
                if s[low]!=s[high]:
                    return False
                low+=1
                high-=1

            return True

        low=0
        high=len(s)-1

        while low<=high:

            if s[low]!=s[high]:
                return check_palindrome(low+1,high) or check_palindrome(low,high-1)
            low+=1
            high-=1
        
        return True
        

File: Data Structures & Algorithms/longest-palindromic-substring/submission-2.js
class Solution {
    /**
     * @param {string} s
     * @return {string}
     */
    longestPalindrome(s) {
        let start=0;
        let max_len=1;
        let n=s.length;
        for(let i=0; i<n; i++){
            let low=i;
            let high=i;

            while (low>=0 && high<n && s[low]==s[high]){
                let curr_len=high-low+1;

                if (curr_len>max_len){
                    start=low;
                    max_len=curr_len;

                }
                low-=1;
                high+=1;
            }

            low=i;
            high=i+1;

            while (low>=0 && high<n && s[low]==s[high]){
                let curr_len=high-low+1;

                if (curr_len>max_len){
                    start=low;
                    max_len=curr_len;

                }
                low-=1;
                high+=1;
            }

        
        }
        return s.substring(start,start+ max_len);
    
    }
}
